code/API_definitions/location-subscriptions.yaml

```diff
@@ -0,0 +1,634 @@
+      
+        In all cases, publicAddress must be specified, along with at least one of either privateAddress or publicPort, dependent upon which is known. In general, mobile devices cannot be identified by their public IPv4 address alone.
+
+    SingleIpv4Addr:
+      type: string
+      description: A single IPv4 address with no subnet mask
+      example: "84.125.93.10"
+      pattern: '^(([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])$'
+    
+    Port:
+      type: integer
+      minimum: 0
+      maximum: 65535
+    
+    Ipv6Address:
+      type: string
+      allOf:
+        - pattern: '^((:|(0?|([1-9a-f][0-9a-f]{0,3}))):)((0?|([1-9a-f][0-9a-f]{0,3})):){0,6}(:|(0?|([1-9a-f][0-9a-f]{0,3})))(\/(([0-9])|([0-9]{2})|(1[0-1][0-9])|(12[0-8])))?$'
+        - pattern: '^((([^:]+:){7}([^:]+))|((([^:]+:)*[^:]+)?::(([^:]+:)*[^:]+)?))(\/.+)?$'
+      example: "2001:db8:85a3:8d3:1319:8a2e:370:7344"
+      description: |
+        IPv6 address, following IETF 5952 format, may be specified in form <address/mask> as:
+          - address - The /128 subnet is optional for single addresses:
+            - 2001:db8:85a3:8d3:1319:8a2e:370:7344
+            - 2001:db8:85a3:8d3:1319:8a2e:370:7344/128
+          - address/mask - an IP v6 number with a mask:
+            - 2001:db8:85a3:8d3::0/64
+            - 2001:db8:85a3:8d3::/64
+    
+    ErrorInfo:
+      type: object
+      required:
+        - status
+        - code
+        - message
+      properties:
+        status:
+          type: integer
+          description: HTTP status code returned along with this error response
+        code:
+          type: string
+          description: Code given to this error
+        message:
+          type: string
+          description: Detailed error description
+    
+    Webhook:
+      type: object
+      description: Detail for event channel
+      properties:
+        notificationUrl:
+          $ref: '#/components/schemas/NotificationUrl'
+        notificationAuthToken:
+          $ref: '#/components/schemas/NotificationAuthToken'
+      required: 
+        - notificationUrl
+    
+    EventSubscriptionId:
+      type: string
+      format: uuid
+      description: EventSubscription ID in UUID format      
+    
+    SubscriptionDetail:
+      type: object
+      description: Subscription detail
+      properties:
+        device:
+          $ref: '#/components/schemas/Device'
+        area:
+          $ref: '#/components/schemas/Area'
+        eventType: 
+          $ref: "#/components/schemas/EventType"
+      required:
+        - device
+        - area
+        - eventType
+    
+    EventType:
+      description: |
+        AREA_ENTERED - Location event triggered when the device entered the given area
+        AREA_LEFT - Location event triggered when the device left the given area
+        SUBSCRIPTION_ENDS - It is inform listener request that subscription terminates.
```

Comment:
@bigludo7 Here we are adding SUBSCRIPTION_ENDS event in the operation to create a subscription and I was checking yesterday's PR from Device Status and we didn't add it there: https://github.com/camaraproject/DeviceStatus/blob/main/code/API_definitions/device-status.yaml#L513

should we align this?

Reply:
Really, I think it make no sense to allow a subscription to SUBSCRIPTION_ENDS event, as this event is triggered when a subscription to another event ends

Reply:
I also think that we don't need that event in the create-subscription, but in the event-notification